clang/test/SemaSYCL/sycl-kernel-entry-point-attr-argument.cpp

```diff
@@ -209,8 +209,40 @@ struct B25_2;
   clang::sycl_kernel_entry_point(B25_2)]]
 void bad25();
 
-struct B26;
-// expected-error@+3 {{'sycl_kernel_entry_point' kernel name argument conflicts with a previous declaration}}
+struct B26 { void operator()() const {} };
+// expected-error@+3 {{'sycl_kernel_entry_point' kernel name 'B26' conflicts with a previous declaration}}
 // expected-note@+1  {{previous declaration is here}}
-[[clang::sycl_kernel_entry_point(B26)]] void bad26_1();
-[[clang::sycl_kernel_entry_point(B26)]] void bad24_2();
+[[clang::sycl_kernel_entry_point(B26)]] void bad26_1(B26 k) { k(); }
+[[clang::sycl_kernel_entry_point(B26)]] void bad24_2(B26 k) { k(); }
+
+
+struct kernel_name;
+
+template<typename KN, typename K>
+[[clang::sycl_kernel_entry_point(KN)]]
+void sycl_entry_point(K ker, int i) { ker(); }
+
+template<typename KN, typename K>
+void sycl_entry_point(K ker, long i) { ker(); }
+
+void proxy() {
```

Comment:
Please follow the existing naming scheme and name these `B27`, `bad27()`, and `test_bad27()`.

A comment would be helpful to indicate that the 2nd overload is intended to validate that incorrect use of the `sycl_kernel_point_attribute` doesn't affect SFINAE and overload resolution.

Reply:
I addressed this in the new patch